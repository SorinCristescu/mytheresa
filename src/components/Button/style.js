import styled from 'styled-components';

export const ButonContainer = styled.div`
  width: 300px;
  padding: 30px auto;
  margin: 20px 0;

  button {
    text-align: center;
    font-family: 'MonumentExtended-Ultrabold';
    font-size: 18px;
    color: white;
    border: none;
    background: transparent;
  }

  button:hover {
    color: white;
    cursor: pointer;
    border-bottom: 2px solid white;
  }

  button:focus {
    outline: none;
  }
`;
