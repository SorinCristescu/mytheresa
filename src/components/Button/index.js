import React from 'react';
import PropTypes from 'prop-types';

// Styles
import { ButonContainer } from './style';

const Button = ({ name, width, onClick }) => {
  return (
    <ButonContainer width={width}>
      <button onClick={onClick}>{name}</button>
    </ButonContainer>
  );
};
// Runtime props type checking
Button.propTypes = {
  name: PropTypes.string.isRequired,
  width: PropTypes.string.isRequired,
  onClick: PropTypes.func,
};
export default Button;
