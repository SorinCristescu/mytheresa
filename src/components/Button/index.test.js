import React from 'react';
import toJSON from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import Button from './index';

it('to render Button component', () => {
  const mockProps = [
    {
      name: 'Click',
      width: '200px',
    },
  ];
  expect(shallow(<Button props={mockProps} />)).toMatchSnapshot();
});
