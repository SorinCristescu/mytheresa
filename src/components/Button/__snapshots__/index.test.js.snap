// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`to render Button component 1`] = `
<styled.div>
  <button />
</styled.div>
`;
